packages/core/src/blueprint-modern.scss

```diff
@@ -0,0 +1,830 @@
+        color: $text-color;
+
+        &:hover {
+          color: $hover-text-color;
+        }
+
+        &:active {
+          color: $active-text-color;
+        }
+      }
+    }
+  }
+
+  // Contrast for default intent
+  color: $white;
+  &.#{$ns}-interactive {
+    &:hover {
+      background: $dark-gray5;
+    }
+
+    &:active {
+      background: $dark-gray4;
+    }
+  }
+
+  .#{$ns}-tag-remove >.#{$ns}-icon:first-child {
+    color: rgba($white, 0.7);
+
+    &:hover {
+      color: $white;
+    }
+
+    &:active {
+      color: $white;
+    }
+  }
+
+  // Dark mode contrast for default intent
+  .#{$ns}-dark & {
+    background: $gray5;
+    color: $pt-text-color;
+
+    &.#{$ns}-interactive {
+      &:hover {
+        color: $gray4;
+      }
+
+      &:active {
+        color: $gray3;
+      }
+    }
+
+    .#{$ns}-tag-remove >.#{$ns}-icon:first-child {
+      color: rgba($dark-gray1, 0.7);
+
+      &:hover {
+        color: $dark-gray1;
+      }
+
+      &:active {
+        color: $dark-gray1;
+      }
+    }
+  }
+
+  // Light mode contrast for intents
+  @each $intent, $color in $tag-intent-colors-modern {
+    &.#{$ns}-intent-#{$intent} {
+      @include pt-tag-modern($color...);
+    }
+  }
+
+  // Minimal style contrast
+  &.#{$ns}-minimal {
+    // Minimal style for default intent
+    background-color: rgba($gray3, 0.15);
+    color: $pt-text-color;
+    &.#{$ns}-interactive {
+      &:hover {
+        background-color: rgba($gray3, 0.3);
+        color: $black;
+      }
+
+      &:active {
+        background-color: rgba($gray3, 0.35);
+        color: $black;
+      }
+
+      .#{$ns}-tag-remove >.#{$ns}-icon:first-child {
+        color: $gray1;
+
+        &:hover {
+          color: $dark-gray1;
+        }
+
+        &:active {
+          color: $dark-gray1;
+        }
+      }
+    }
+
+    // Minimal dark style for default intent
+    .#{$ns}-dark & {
+      background-color: rgba($gray3, 0.15);
+      color: $light-gray1;
+
+      &.#{$ns}-interactive {
+        &:hover {
+          background-color: rgba($gray3, 0.3);
+          color: $white;
+        }
+
+        &:active {
+          background-color: rgba($gray3, 0.35);
+          color: $white;
+        }
+      }
+
+      .#{$ns}-tag-remove >.#{$ns}-icon:first-child {
+        color: $gray4;
+
+        &:hover {
+          color: $light-gray1;
+        }
+
+        &:active {
+          color: $light-gray1;
+        }
+      }
+    }
+
+    @each $intent, $color in $minimal-tag-intent-colors-modern {
+      &.#{$ns}-intent-#{$intent} {
+        @include pt-tag-minimal-intent-modern($color...);
+      }
+    }
+
+    @each $intent, $color in $minimal-dark-tag-intent-colors-modern {
+      &.#{$ns}-intent-#{$intent} {
+        @include pt-tag-minimal-dark-intent-modern($color...);
+      }
+    }
+  }
+}
+
+// Contrast for toasts
+.#{$ns}-toast {
+  // HACK needs !important to override the existing hack
+  /* stylelint-disable declaration-no-important */
```

Comment:
nit: does this only disable the rule for this Sass block? do we need to re-enable it?

Reply:
ah yes, it needs to re-enabled